### ISSUE TITLE
clarification on the BERT vocab

### ISSUE THREAD

**Issue by vr25** [xhan77/AdaptaBERT#1]
Hello,

I read the paper and I have one quick question regarding the vocabulary. When BERT is domain fine-tuned, this is what my understanding is. Take the BERT pre-trained model (say, bert-base-uncased) which comes with its own config, vocab and the model. When you use this pre-trained model on your domain-specific corpus, all you're basically doing is extending the above base model by training it on additional domain-specific documents (following the same (1) masked LM and (2) NSP tasks).

It is not quite clear from the paper if the vocab is extended too for the OOV words in the new domain corpus and are the existing words in the vocab retrained too.

Similarly, does the vocab for task fine-tuning get modified or it is just that the model learns new weights?

Kindly let me know if I am missing something.

Thank you.

**Comment by xhan77** [comment]
Thanks for your interest in our paper! I'll put my reply in email here.

So first, BERT uses a wordpiece style tokenizer, which makes “real” OOV words (<UNK>) really rare. For example, “hym” (him) in historical English would be represented with “hy” and “##m” when being passed into the model, and both “hy” and “##m” are in the original vocab set. Therefore, we do not have a different vocab for the new domain corpus. All the fine-tuning, both domain-purpose and task-purpose, would be applied on the original vocab set, so the embeddings for words/wordpieces in the vocab will change (whereas the vocab would still include the same set of words/wordpieces). Also, just a minor clarification, our domain-adaptive tuning doesn’t use the NSP objective since it’s not always available for the new domain corpus.
